catalyst-gateway/bin/src/event_db/voter_registration.rs

```diff
@@ -38,17 +42,104 @@ impl EventDB {
                 &[
                     &hex::decode(tx_id).map_err(|e| Error::DecodeHex(e.to_string()))?,
                     &stake_credential,
-                    &public_voting_key,
+                    &public_voting_key.as_bytes(),
                     &payment_address,
                     &nonce,
-                    &metadata_61284,
-                    &metadata_61285,
+                    &metadata_cip36,
+                    &report,
                     &valid,
-                    &stats,
                 ],
             )
             .await?;
 
         Ok(())
     }
+
+    /// Index registration data
+    pub async fn index_registration_data(
+        &self, txs: Vec<MultiEraTx<'_>>, slot_no: SlotNumber, network: Network,
+    ) -> Result<(), Error> {
+        let cddl = CddlConfig::new();
+
+        for tx in txs {
+            let mut valid_registration = true;
+
+            if !tx.metadata().is_empty() {
```

Comment:
Is it necessary to make this check here as we are passing this stuff to the function ?

Reply:
yea its not very clean but this allows to just ignore the tx and move to the next one if the metadata is empty. We just discard it. We could move it further down the call stack but it gets messy.